src/wast-parser.cc

```diff
@@ -1313,11 +1314,24 @@ Result WastParser::ResolveRefTypes(const Module& module,
 
     assert(type.IsReferenceWithIndex());
 
+    if (ref_var.var.is_index()) {
+      if (type.GetReferenceIndex() >= module.types.size()) {
```

Comment:
Is `module.types` complete at this point? i.e. might it not be possible that there are more types to be be parsed?

Reply:
Yes, all names are resolved after the module is fully parsed. The implicit function types has not been added yet.